src/Common/src/System/Security/Cryptography/RSACng.cs

```diff
@@ -55,6 +66,13 @@ protected override byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm)
         {
             return CngCommon.HashData(data, hashAlgorithm);
         }
+
+        private void SetKeySize(int newKeySize)
+        {
+            _skipKeySizeCheck = true;
+            KeySize = newKeySize;
```

Comment:
Can KeySize throw?  I'm wondering if the resetting of _skipKeySizeCheck should be in a finally.